persistence-api/src/main/java/io/stargate/db/datastore/DataStoreFactory.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright DataStax, Inc. and/or The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.datastore;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Strings;
+import io.stargate.db.AuthenticatedUser;
+import io.stargate.db.ClientInfo;
+import io.stargate.db.Persistence;
+import java.net.InetSocketAddress;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class DataStoreFactory {
+
+  private final Persistence persistence;
+
+  /**
+   * @param persistence the persistence to use for querying (some methods effectively creates a new
+   *     {@link Persistence.Connection} underneath).
+   */
+  public DataStoreFactory(Persistence persistence) {
+    this.persistence = persistence;
+  }
+
+  /**
+   * Creates a new DataStore using the provided connection for querying and with the provided
+   * default parameters.
+   *
+   * @param connection the persistence connection to use for querying.
+   * @param options the options for the create data store.
+   * @return the created store.
+   */
+  public DataStore create(Persistence.Connection connection, @Nonnull DataStoreOptions options) {
```

Comment:
Now that `Persistence` is encapsulated in `DataStoreFactory`, exposing a `public` method that takes a `Persistence.Connection` parameter seems contradictory to me.

Reply:
Good point, I've made it private.